CONTRIBUTING.md

```diff
@@ -1,51 +1,147 @@
+
+1. Press <kbd>Ctrl+Shift+D</kbd> to open the Run view in VS Code and ensure `Launch Extension` is selected.
+2. Start debugging by pressing <kbd>F5</kbd>.
+
+#### Testing
+
+To run tests:
+
+1. Execute `npm run test` or press <kbd>F5</kbd> in VS Code with the "Launch Tests" debug configuration selected.
+2. For integration tests, select `Razor Integration Tests` and press <kbd>F5</kbd> to start debugging.
+
+### Setting Up Local Language Servers
+
+This section shows how to set up local Razor or Roslyn language servers for debugging with the VSCode C# extension.
+
+#### Roslyn
+
+1. Clone the [Roslyn repository](https://github.com/dotnet/roslyn). This repository contains the Roslyn server implementation.
+2. Follow the build instructions provided in the repository.
+
+The server DLL is typically at `$roslynRepoRoot/artifacts/bin/Microsoft.CodeAnalysis.LanguageServer/Debug/net7.0/Microsoft.CodeAnalysis.LanguageServer.dll`, but this may vary based on the built configuration.
+
+#### Razor
+
+1. Clone the [Razor repository](https://github.com/dotnet/razor). This repository contains the Razor server implementation.
+2. Follow the build instructions provided in the repository.
+
+The server DLL is typically at `$razorRepoRoot/artifacts/bin/rzls/Debug/net8.0`.
+
+### Configuring Local Language Servers
+
+This section provides instructions on how to debug locally built Roslyn and Razor language servers. You can do this by either directly editing the `settings.json` file or through the VSCode settings interface.
+
+#### Roslyn
+
+1. Navigate to `C:\Users\YourUsername\AppData\Roaming\Code\User\settings.json` and add the following lines:
+
+```json
+"dotnet.server.waitForDebugger": true,
+"dotnet.server.path": "$roslynRepoRoot\\artifacts\\bin\\Microsoft.CodeAnalysis.LanguageServer\\Debug\\net7.0\\Microsoft.CodeAnalysis.LanguageServer.dll"
+```
 
-To run and develop the extension do the following:
+Replace `$roslynRepoRoot` and `YourUsername` with your actual values.
 
-* Run `npm i`
-* Run `npm i -g gulp`
-* Run `gulp installDependencies` (this will download the various dependencies as specified by the version in the [package.json](package.json))
-* Open in Visual Studio Code (`code .`)
-* _Optional:_ run `npm run watch`, make code changes
-* Press <kbd>F5</kbd> to debug
+Or, in VSCode settings (`Ctrl+,`):
 
-To **test** do the following: `npm run test` or <kbd>F5</kbd> in VS Code with the "Launch Tests" debug configuration.
+1. Search for `dotnet server`.
+2. Set `dotnet.server.path` to the path of your Roslyn DLL.
+3. Enable `dotnet.server.waitForDebugger`.
 
-### Using a locally developed Roslyn server
+#### Razor
 
-https://github.com/dotnet/roslyn contains the server implementation.  Follow the instructions there to build the repo as normal.  Once built, the server DLL will be located in the build output directory, typically
+1. Navigate to `C:\Users\YourUsername\AppData\Roaming\Code\User\settings.json` and add the following lines:
```

Comment:
Aside from the fact that these are Windows specific paths, I think it's better to tell people to create a settings.json in the folders they want to use for debugging VS Code, or use the "Workspace" tab in the settings UI when changing settings

Reply:
I just updated the PR to used the "workspace" tab :)

Reply:
okay I just updated the windows specific paths